selfdrive/car/volkswagen/values.py

```diff
@@ -53,6 +54,11 @@ class CAR:
   GOLF = "VOLKSWAGEN GOLF"
   AUDI_A3 = "AUDI A3"
 
+MQB_CARS = {
+  CAR.GOLF,                 # Chassis AU, 2013-2020, includes Golf, Alltrack, Sportwagen, GTI, GTI TCR, GTE, GTD, Clubsport, Golf R, e-Golf
```

Comment:
let's remove this list for the time being. I could easily see a not-obvious bug where we merge a car port that forgets to put the new car in the list and ends up with no car params set

Reply:
I request that we keep it.

The `candidate in MQB_CARS` block in interface.py is designed to collect all the MQB-specific work together, and allow the same for the PQ-specific work. It may be a long time (or never) before Comma accepts that work upstream. However, the task of maintaining a PQ development fork will involve a much larger diff with much more frequent merge conflicts if this part isn't included now.

TL;DR: Please don't make my job harder. :)

Since you requested process_replay drives for every make/model (and we intend to provide them) I think that would suffice for now. Beyond that, I think every other car port already lets get_params() run off the end with no data if there's no match. Robustly dealing with that is a laudable goal but this PR isn't really the vehicle to make it happen.

Reply:
we need routes for test car models, but process replay wouldn't catch that anyway. process replay assumes that you start in a known, good state.

I understand wanting to minimize the diff. I propose we use simply use one list (PQ_CARS, MQB gets privileges for being first). See honda for precedent on this: https://github.com/commaai/openpilot/blob/master/selfdrive/car/honda/values.py#L1160.

Reply:
I think one list is preferably anyway. We'll run into the same thing where we have to make sure each car is in exactly one of the lists.

Reply:
I'm not sure that proposal advances either of our goals.

Your stated concern is "_a not-obvious bug where we merge a car port that forgets to put the new car in the list and ends up with no car params set_". The proposal makes the penalty for "forgetting" into a _wrong_ carParams situation rather than _no_ carParams, which is a large backward step in obviousness and safety.

My equivalent of the [Honda Bosch conditional section of interface.py](https://github.com/commaai/openpilot/blob/d72855ecb8749ca994af4f1b5cf1b575c068e88e/selfdrive/car/honda/interface.py#L127-L136) would be considerably longer (for trans type and network location detection) and would be a 100% diff just due to indentation, unless you let me hang it all behind an `if True:` or something until a later possible merge of PQ support.

Your point about making sure each CAR is on exactly one list is well-taken, but `test_car_interfaces.py` could be augmented to perform that check.

Reply:
I'm fine with an `if True:` for the time being. I'd just really like to avoid dead code in the hopes of one day upstreaming something which isn't in a very upstreamable state.

Reply:
Fair enough, I'll make the changes shortly. Thanks for the accommodation.

Reply:
Done. And yes, I'd like to retain the comments for what fits into each CAR/chassis type. Otherwise, people adding new FPv2 data will try to add new CARs for e.g., the dozen variants of VW Golf, which is a bad idea for (wall of text).